Uninstallation.md

```diff
@@ -30,63 +30,94 @@ remove-->
 If you also intend to delete the Chocolatey directory, remove the `-WhatIf`:
 
 ~~~powershell
-if (!$env:ChocolateyInstall) {
-  Write-Warning "The ChocolateyInstall environment variable was not found. `n Chocolatey is not detected as installed. Nothing to do"
-  return
+$VerbosePreference = 'Continue'
+if (-not $env:ChocolateyInstall) {
+    $message = @(
+        "The ChocolateyInstall environment variable was not found."
+        "Chocolatey is not detected as installed. Nothing to do."
+    ) -join "`n"
+
+    Write-Warning $message
+    return
 }
-if (!(Test-Path "$env:ChocolateyInstall")) {
-  Write-Warning "Chocolatey installation not detected at '$env:ChocolateyInstall'. `n Nothing to do."
-  return
+if (-not (Test-Path "$env:ChocolateyInstall")) {
+    $message = @(
+        "No Chocolatey installation detected at '$env:ChocolateyInstall'."
+        "Nothing to do."
+    ) -join "`n"
+
+    Write-Warning $message
+    return
 }
 
-$userPath = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey('Environment').GetValue('PATH', '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames).ToString()
-$machinePath = [Microsoft.Win32.Registry]::LocalMachine.OpenSubKey('SYSTEM\CurrentControlSet\Control\Session Manager\Environment\').GetValue('PATH', '', [Microsoft.Win32.RegistryValueOptions]::DoNotExpandEnvironmentNames).ToString()
+$userPath = [Microsoft.Win32.Registry]::CurrentUser.OpenSubKey(
+    'Environment'
+).GetValue('PATH', '', 'DoNotExpandEnvironmentNames').ToString()
```

Comment:
I know the original code used .NET here but is there a reason the native PowerShell cmdlets are not being used?

Reply:
I should add a comment explaining this, for sure. The gist of it is that the PowerShell registry provider doesn't really expose a way to retrieve values _without_ expanding the environment variables.

Because of this, if we use the registry provider from PowerShell we will be accidentally replacing the PATH values with absolute values, instead of incorporating the original environment variables; if a PATH value is `%WINDIR%\config` then PS's registry provider sees `C:\Windows\Config` and we may end up corrupting the PATH variables we're trying to preserve.